src/libmongoc/src/mongoc/mongoc-rpc.c

```diff
@@ -993,6 +997,29 @@ _mongoc_rpc_get_first_document (mongoc_rpc_t *rpc, bson_t *reply)
    return false;
 }
 
+/* Get the first BSON document from an OP_MSG reply: */
+bool
+_mongoc_rpc_reply_get_first_msg (mongoc_rpc_msg_t *reply_msg,
+                                 bson_t *bson_reply)
+{
+   /* Note that mongo_rpc_reply_t is a union, with a mongo_rpc_msg_t field; see
+   the *.def files and MongoDB Wire Protocol documentation for details. */
+
+   int32_t document_len;
```

Comment:
Good catch! Although now there's a question of which is right-- all the .def files use int32, but below we see:

   document_len = BSON_UINT32_FROM_LE (document_len);

...while I think uint32 is /logical/, in _mongoc_rpc_reply_get_first() there's the same behavior! Is this a bug?

@kevinAlbs Any thoughts here?

Reply:
On [bsonspec.org](https://bsonspec.org/spec.html) the four byte document length is specified as a signed int32. I think that is why the .def files use int32. AFAIK the expectation is that the length is non-negative. Otherwise it is invalid BSON.

Reply:
It probably is better to fix them in a batch. Agreed that signed to unsigned is defined (although can still be "fun"!). I will note this for CDRIVER-4280. Thanks everyone for the comments!